Code/Tools/ProjectManager/Source/PythonBindings.cpp

```diff
@@ -1244,11 +1263,12 @@ namespace O3DE::ProjectManager
 
         if (!result.IsSuccess())
         {
-            return result;
+            AZStd::pair<AZStd::string, AZStd::string> pythonRunError(result.GetError(), result.GetError());
+            return AZ::Failure<AZStd::pair<AZStd::string, AZStd::string>>(AZStd::move(pythonRunError));
         }
         else if (!downloadSucceeded)
         {
-            return AZ::Failure<AZStd::string>("Failed to download gem.");
+            return AZ::Failure<AZStd::pair<AZStd::string, AZStd::string>>(GetSimpleDetailedErrorPair());
```

Comment:
We use `AZStd::pair<AZStd::string, AZStd::string>` enough, it might be nice to add a `using AZStd::pair<AZStd::string, AZStd::string> ErrorPair;` type in PythonBindingsInterface to simplify.